packages/theme-language-server-common/src/renamed/RenameHandler.ts

```diff
@@ -18,10 +19,14 @@ export class RenameHandler {
   private handlers: BaseRenameHandler[];
   constructor(
     connection: Connection,
+    capabilities: ClientCapabilities,
     private documentManager: DocumentManager,
     private fileExists: FileExists,
   ) {
-    this.handlers = [new SnippetRenameHandler(connection), new AssetRenameHandler(connection)];
+    this.handlers = [
+      new SnippetRenameHandler(connection, capabilities),
+      new AssetRenameHandler(connection, capabilities),
+    ];
```

Comment:
Instead of passing it into the specific handlers, couldn't we just add the `if (!this.capabilities.hasApplyEditSupport) return;` block inside of `onDidRenameFiles` below?

This would reduce the complexity of the individual handlers.